src/components/application_manager/src/state_controller_impl.cc

```diff
@@ -791,31 +797,33 @@ void StateControllerImpl::OnApplicationRegistered(
   OnStateChanged(app, initial_state, new_state);
 }
 
-int64_t StateControllerImpl::SendBCActivateApp(
+int64_t StateControllerImpl::SendStateToHMI(
     ApplicationConstSharedPtr app,
     hmi_apis::Common_HMILevel::eType level,
     bool send_policy_priority) {
   LOG4CXX_AUTO_TRACE(logger_);
-  smart_objects::SmartObjectSPtr bc_activate_app_request =
-      MessageHelper::GetBCActivateAppRequestToHMI(
-          app,
-          app_mngr_.connection_handler().get_session_observer(),
-          app_mngr_.GetPolicyHandler(),
-          level,
-          send_policy_priority,
-          app_mngr_);
-  if (!bc_activate_app_request) {
-    LOG4CXX_ERROR(logger_, "Unable to create BC.ActivateAppRequest");
+  smart_objects::SmartObjectSPtr request = NULL;
+  if (level == hmi_apis::Common_HMILevel::NONE) {
+    request = MessageHelper::GetBCCloseApplicationRequestToHMI(app, app_mngr_);
+  } else {
+    request = MessageHelper::GetBCActivateAppRequestToHMI(
+        app,
+        app_mngr_.GetPolicyHandler(),
+        level,
+        send_policy_priority,
+        app_mngr_);
+  }
+  if (!request) {
+    LOG4CXX_ERROR(logger_, "Unable to create request");
     return -1;
   }
-  if (!app_mngr_.GetRPCService().ManageHMICommand(bc_activate_app_request)) {
-    LOG4CXX_ERROR(logger_, "Unable to send BC.ActivateAppRequest");
+  if (!app_mngr_.GetRPCService().ManageHMICommand(request)) {
+    LOG4CXX_ERROR(logger_, "Unable to send request");
     return -1;
   }
-  const int64_t corr_id =
-      (*bc_activate_app_request)[strings::params][strings::correlation_id]
-          .asInt();
-  return corr_id;
+  const uint32_t corr_id =
```

Comment:
Why the change to uint32 and then recast to int64?

Reply:
It seems this was originally chosen for the return value of `SendBCActivateApp` to allow for an error response of `-1`, correlation IDs use uint32_t. I could remove these casts, but it was being done implicitly before anyway.